beacon-chain/light-client-network.md

```diff
@@ -0,0 +1,179 @@
+
+All data types are specified in light client [sync protocol](https://github.com/ethereum/consensus-specs/blob/dev/specs/altair/light-client/sync-protocol.md#containers).
+
+#### Retrieval
+
+The network supports the following mechanisms for data retrieval:
+
+* `LightClientBootstrap` structure by a post-Altair beacon block root.
+* `LightClientUpdatesByRange` - requests the `LightClientUpdate` instances in the sync committee period range [start_period, start_period + count), leading up to the current head sync committee period as selected by fork choice.
+* The latest `LightClientFinalityUpdate` known by a peer.
+* The latest `LightClientOptimisticUpdate` known by a peer.
+
+## Specification
+
+### Distance Function
+
+The beacon chain light client network uses the stock XOR distance metric defined in the portal wire protocol specification.
+
+### Content ID Derivation Function
+
+The beacon chain light client network uses the SHA256 Content ID derivation function from the portal wire protocol specification.
+
+### Wire Protocol
+
+The [Portal wire protocol](./portal-wire-protocol.md) is used as the wire protocol for the Beacon Chain Light Client network.
+
+#### Protocol Identifier
+
+As specified in the [Protocol identifiers](./portal-wire-protocol.md#protocol-identifiers) section of the Portal wire protocol, the `protocol` field in the `TALKREQ` message **MUST** contain the value of `0x501A`.
+
+#### Supported Messages Types
+
+The beacon chain light client network supports the following protocol messages:
+
+- `Ping` - `Pong`
+- `Find Nodes` - `Nodes`
+- `Find Content` - `Found Content`
+- `Offer` - `Accept`
+
+#### `Ping.custom_data` & `Pong.custom_data`
+
+In the beacon chain light client network the `custom_payload` field of the `Ping` and `Pong` messages is the serialization of an SSZ Container specified as `custom_data`:
+
+```
+custom_data = Container(data_radius: uint256)
+custom_payload = serialize(custom_data)
+```
+
+### Routing Table
+
+The Beacon CHain Light Client Network uses the standard routing table structure from the Portal Wire Protocol.
+
+### Node State
+
+#### Data Radius
+
+The Beacon Chain Light Client Network includes one additional piece of node state that should be tracked. Nodes must track the `data_radius`
+from the Ping and Pong messages for other nodes in the network. This value is a 256 bit integer and represents the data that
+a node is "interested" in.
+
+We define the following function to determine whether node in the network should be interested in a piece of content:
+
+```
+interested(node, content) = distance(node.id, content.id) <= node.radius
+```
+
+A node is expected to maintain `radius` information for each node in its local node table. A node's `radius` value may fluctuate as the contents of its local key-value store change.
+
+A node should track their own radius value and provide this value in all Ping or Pong messages it sends to other nodes.
+
+### Data Types
+
```

Comment:
Is the mechanism to distinguish between different networks separate? e.g., mainnet vs goerli? How about individual forks? Maybe needs to incorporate `fork_digest` in the content keys here.

Reply:
We don't currently have any plans to support different networks.  Portal network is end-user facing, and requires a critical mass of nodes to have sufficient capacity to serve the necessary data.  Between these two things we A: don't have a compelling use case to motivate support for test networks and B: don't think the network would be reliable for test networks since there would likely not be sufficient nodes to serve the underlying data.